### ISSUE TITLE
Query string in repo URL is discarded when downloading packages

### ISSUE THREAD

**Issue by tomasr** [helm/helm#4419]
Helm version: 
```text
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

Ran into [this](https://cwienczek.com/setting-up-secure-helm-chart-repository-on-azure-blob-storage/) article which suggests it is possible to use an Azure Blob Storage account as a Helm repository, while using SAS Tokens to authenticate the requests to the storage account.

This worked based on the fix for [this issue](https://github.com/helm/helm/issues/2937). However, seems like this broke after that.

Helm is able to download the index.yaml file successfully, but then fails to download the chart's `.tgz` file with a 404 error.

Looking through the history, I believe this broke during [this change](https://github.com/helm/helm/commit/af9190f956cf8983370fe6b6d96aa30e9552c94a#diff-3c52d2d00ef683b18b7c7577450a25bd). to return absolute URLs from `FindChartInRepoURL`.

Sounds like the right solution would be to make sure that the query string in the repository URL is copied over to the absolute chart URL in `MakeAbsoluteChartURL`.

**Comment by bacongobbler** [comment]
Can you describe the steps or perhaps provide a unit test that reproduces the bug?

**Comment by tomasr** [comment]
Submitted PR with additional tests showing the issue + first attempt at a fix https://github.com/helm/helm/pull/4438.

**Comment by thomastaylor312** [comment]
I was able to duplicate the bug with the test case @tomasr provided in #4438. Pulling this into 2.10

**Comment by bacongobbler** [comment]
fixed in #4438
